mvcc/kvstore.go

```diff
@@ -402,11 +407,20 @@ func (s *store) restore() error {
 		// restore index
 		switch {
 		case isTombstone(key):
-			s.kvindex.Tombstone(kv.Key, rev)
+			if ki, ok := unordered[string(kv.Key)]; ok {
+				ki.tombstone(rev.main, rev.sub)
+			}
 			delete(keyToLease, string(kv.Key))
 
 		default:
-			s.kvindex.Restore(kv.Key, revision{kv.CreateRevision, 0}, rev, kv.Version)
```

Comment:
Not removing it in this PR because TestIndexCompact and TestIndexRestore in index_test.go are using it. So if we remove it we need to modify these test cases accordingly. Hence better to raise it in separate PR once this gets merged.